pinset.go

```diff
@@ -60,21 +81,55 @@ func NewPinRequest(path string, privKey crypto.PrivKey, addrs []string) (*PinReq
 // MemPinset is a completely ficticious implementation of a pinset
 // it shouldn't be used anywhere, by anyone, ever.
 type MemPinset struct {
+	pk       PinStatusStore
 	Profiles Profiles
-	Pins     []string
+	pins     []string
 }
 
 // Pin a dataset
-func (m *MemPinset) Pin(req *PinRequest) (chan Progress, error) {
-	m.Pins = append(m.Pins, req.Path)
-	return nil, nil
+func (m *MemPinset) Pin(req *PinRequest) (chan PinStatus, error) {
+	if len(m.pins) == 0 {
+		m.pins = append(m.pins, req.Path)
+	} else {
+		for i, p := range m.pins {
```

Comment:
The intention here is to store the pins in reverse order? This won't work if the Path is less than any other value in the sorted list. For example:

```
m.pins = ["z", "m", "b"]
Pin({Path: "a"})
```

During the iteration, req.Path is never `>` than p, so it'll never be inserted.

Making a function just for insertion, like the one given [here](https://gist.github.com/zhum/57cb45d8bbea86d87490#gistcomment-2276438), seems better. And it can be directly tested.